components/faction/commons/faction.lua

```diff
@@ -35,56 +48,98 @@ Faction.types.FactionProps = TypeUtil.struct{
 	faction = Faction.types.Faction,
 }
 
-local byName = Table.map(Data.factionProps, function(faction, props) return props.name, faction end)
-local byLowerName = Table.map(byName, function(name, faction) return name:lower(), faction end)
+
+local factionsByName = FnUtil.memoize(function ()
+	return Table.mapValues(Data.factionProps, function(factionProps)
+		return Table.map(factionProps, function(faction, props) return props.name:lower(), faction end)
+	end)
+end)
+
+---Parses the option table, eventually adding defaults
+---@param options table?
+---@return table
+function Faction._parseOptions(options)
+	if type(options) ~= 'table' then
+		options = {}
+	end
+
+	options.game = options.game or Data.defaultGame
+	return options
+end
+
+---Returns a list of valid factions
+---@param options {game: string?}?
+---@return string[]
+function Faction.getFactions(options)
+	options = Faction._parseOptions(options)
+	local game = options.game
+	return game and Data.factions[game] or {}
+end
+
+---Returns a list of valid faction aliases
+---@param options {game: string?}?
+---@return string[]
+function Faction.getAliases(options)
+	options = Faction._parseOptions(options)
+	local game = options.game
+	return game and Data.aliases[game] or {}
+end
 
 --- Checks if a entered faction is valid
 ---@param faction string?
+---@param options {game: string?}?
 ---@return boolean
-function Faction.isValid(faction)
-	return Data.factionProps[faction] ~= nil
+function Faction.isValid(faction, options)
+	options = Faction._parseOptions(options)
+	local game = options.game
+	return game and (Data.factionProps[game] or {})[faction] ~= nil
```

Comment:
maybe could make it one line
```suggestion
	return game and (Data.factionProps[Faction._parseOptions(options).game] or {})[faction] ~= nil
```

Reply:
wouldn't work as game is used first, would need it as
```suggestion
	local game = Faction._parseOptions(options).game
	return game and (Data.factionProps[game] or {})[faction] ~= nil
```

Reply:
oh true, and yeah lets us save a few lines of code